packages/amplify-schema-validator/src/helpers/is-scalar-or-enum.ts

```diff
@@ -49,7 +49,7 @@ export const isScalarOrEnum = (
     return isScalarOrEnum(type.type, enums);
   } if (type.kind === Kind.LIST_TYPE) {
     /* istanbul ignore next */
-    return isScalarOrEnum(type.type, enums);
+    return false;
```

Comment:
Index should be non scalar validation is failing for list type. Return false directly for list type fields

Reply:
LGTM.